components/button/button-add.js

```diff
@@ -56,49 +74,28 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 				width: 100%;
 			}
 			button:hover .line,
-			button:${unsafeCSS(getFocusPseudoClass())} .line {
+			button:focus .line {
 				border-top-color: var(--d2l-color-celestine);
 			}
 
-			.content {
-				align-items: center;
-				background-color: white;
-				display: flex;
-				position: absolute;
-			}
-			:host([text-visible]) .content {
-				color: var(--d2l-color-celestine);
-				height: 1.5rem;
-				padding: 0 0.3rem;
-			}
-
-			:host([text-visible]) d2l-icon,
-			:host(:not([text-visible])) button:hover d2l-icon,
-			:host(:not([text-visible])) button:${unsafeCSS(getFocusPseudoClass())} d2l-icon {
-				color: var(--d2l-color-celestine);
-			}
-			:host(:not([text-visible])) d2l-icon {
-				color: var(--d2l-color-galena);
-				margin: -3px; /** hover/click target */
-				padding: 3px; /** hover/click target */
-			}
-			:host([text-visible]) d2l-icon {
-				padding-inline-end: 0.2rem;
+			button:hover d2l-button-add-icon-text,
+			button:focus d2l-button-add-icon-text {
+				--d2l-button-add-icon-text-icon-color: var(--d2l-color-celestine);
 			}
-
-			span {
-				font-size: 0.7rem;
-				font-weight: 700;
-				letter-spacing: 0.2px;
-				line-height: 1rem;
+			:host([visibility-condition="nearby"]) button:not(:focus):not(:hover) d2l-button-add-icon-text,
+			:host([visibility-condition="hover-focus"]) button:not(:focus):not(:hover) d2l-button-add-icon-text {
```

Comment:
In order to not have the background color issue, there are now 2 lines. When `button-add` is not hovered on or focused on and the visibility condition is nearby or hover (i.e., inner content should not be visible), positioning the inner content absolute allows the 2 lines to combine

Reply:
Neat!